### PR TITLE
Ensure the path is case sensitive

### LINKED ISSUE #9: listContents on subfolder of camelcase folder returns nothing
HI!

This bug seems similar to: https://github.com/thephpleague/flysystem-dropbox/issues/3

Dropbox API v1 and v2 are both case-insensitive: https://www.dropboxforum.com/t5/API-support/Case-Sensitivity-in-API-2/m-p/191279#M8337

If I try to fetch "Test1/Test2/toto.txt" that returns nothing, because Dropbox returns "test1/Test2/toto.txt" and does not match.

Can we do somthing for that?

Thanks! :)

### PR DESCRIPTION
**Pull request by Alex-D**
In order to fix #9 I've made a patch which can be down to dropbox-api level, but it can do some more calls to Dropbox and only do some failures with Flysystem ecosystem.

### CODE REVIEW

File: src/DropboxAdapter.php [hunk 1]
```diff
@@ -182,11 +182,39 @@ public function listContents($directory = '', $recursive = false): array
             return [];
         }
 
-        return array_map(function ($entry) {
+        $cleanedPathDisplay = [];
```
Comment (freekmurze) [inline src/DropboxAdapter.php]: Generally speaking I don't look for loops and temporary variables. Could you refactor this code to use `array_map` and related functions?

Comment (freekmurze) [inline src/DropboxAdapter.php]: This is quite a long function. And to me it's isn't entirely clear what is happening. Your first reflex might be to add some comments here and there to make things more clear. But what I would prefer is that this block of code would be split up in multiple smaller functions that can be covered by unit tests.

**Comment by freekmurze** [comment]
Don't know for certain if this going to be pulled in, but I left some comments on the code.

**Comment by Alex-D** [comment]
Can't use array_map or similar because I create a *new associative array*.

I've added a lot of comments, hope it helps you to understand my code :)

**Comment by freekmurze** [comment]
I really don't like the fact that the function is so long. Could you break it up into other function. Also make that function public and write some tests specifically for the function. That'll help me to further refactor the code.

**Comment by Alex-D** [comment]
https://github.com/spatie/flysystem-dropbox/pull/10/commits/0eac2f9bdbfa84619997ef45600401e0ee89feb0#diff-ffb6bf1e29116d4ede50555132030d1aR307

You want I split that? oO

**Comment by freekmurze** [comment]
> And why this function should be public?

To make it easier to test.

> Should I split that?

Take a first stab at it. I'll refactor if necessary.

**Comment by freekmurze** [comment]
I'll refactor it myself. Thank you for this!

**Comment by Alex-D** [comment]
Sorry for the lack of time ^^'

File: src/DropboxAdapter.php [hunk 2]
```diff
@@ -293,4 +302,54 @@ protected function normalizeResponse(array $response): array
...
     }
+
+    protected function getCleanedPathDisplay($entries)
+    {
+        // init temp associative array that will contains
+        // path lower as key and path display as value
+        $cleanedPathDisplay = [];
+        foreach ($entries as $entry) {
+            // we only need folder paths
+            if ($entry['.tag'] === 'folder') {
```
Comment (freekmurze) [inline src/DropboxAdapter.php]: Why only folders here? Doesn't this problem apply to files as well?

Comment (Alex-D) [inline src/DropboxAdapter.php]: Files are always returned with the good case by Dropbox :)

**Pull request merged**